src/bindings/python/src/openvino/frontend/jax/jaxpr_decoder.py

```diff
@@ -0,0 +1,288 @@
+
+import jax.core
+from openvino.frontend.jax.py_jax_frontend import _FrontEndJaxDecoder as Decoder
+from openvino.runtime import PartialShape, Type as OVType, OVAny
+from openvino.frontend.jax.utils import jax_array_to_ov_const, get_ov_type_for_value, \
+    ivalue_to_constant
+
+import jax
+
+from typing import List
+import logging
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.WARNING)
+
+class JaxprPythonDecoder (Decoder):
+    '''
+    The jaxpr decoder uses Jaxpr to get graph information from a jax module.
+    It takes use of the following parts.
+    
+    - `ClosedJaxpr`: the jaxpr object that contains the jaxpr and literals.
+        - `Jaxpr`: the jaxpr object that contains the invars, outvars, and eqns.
+            - `JaxEqns`: A list of jaxpr equations, which contains the information of the operation.
+                - `Primitive`: the operation that is used in the equation.
+                - `invars`: the input variables of the equation.
+                    - `aval`: the abstract value.
+                - `outvars`: the output variables of the equation.
+                    - `aval`: the abstract value.
+            - `invars`: the input variables of the equation.
+                - `aval`: the abstract value.
+            - `outvars`: the output variables of the equation.
+                - `aval`: the abstract value.
+            - `constvars`: the constant variables.
+                - `aval`: the abstract value.
+        - `Literal`: the literal object that contains the value of the constants.
+    '''
+    
+    def __init__(self, jaxpr, name=None, literals=None):
+        '''
+        Inputs: 
+            - jaxpr: for users, `ClosedJaxpr` is expected here. See https://github.com/google/jax/blob/jaxlib-v0.4.29/jax/_src/core.py#L197
+            - name: the name for the model.
+            - literals: the literals (constants) that are used in the model.
+        '''
+        Decoder.__init__(self)
+        
+        if isinstance(jaxpr, (jax.core.JaxprEqn, jax.core.Jaxpr, jax.core.Var)):
+            self.jaxpr = jaxpr
+        elif isinstance(jaxpr, jax.core.ClosedJaxpr):
+            # Take the `Jaxpr` from `ClosedJaxpr`, see https://github.com/google/jax/blob/jaxlib-v0.4.29/jax/_src/core.py#L85
+            self.jaxpr = jaxpr.jaxpr
+            # Literal should be a `Jax.core.Var`, see https://github.com/google/jax/blob/jaxlib-v0.4.29/jax/_src/core.py#L85
+            self.literals = jaxpr.literals
+        else:
+            raise ValueError(f"Unexpected type of jaxpr: {type(jaxpr)}")
+        self.name = name
+        if self.name is None:
+            self.name = "jax_module"
+        if literals is not None:
+            self.literals = literals
+            
+        self.params = {}
+        if hasattr(self.jaxpr, 'params') and isinstance(self.jaxpr.params, dict):
+            for k in self.jaxpr.params.keys():
+                self.params[k] = self.convert_param_to_constant_node(self.jaxpr, k)
```

Comment:
named param seems to be similar to named input in FX decoder. I don't completely like how it was done in FX decoder, we had to align API with TS in some way, maybe you could do better since you don't have other decoders to align with? Can it be non-constant? For example output of another operation?

Reply:
I have checked nearly all the functions in `jax.lax` and didn't find such case, though I'm not 100% sure that this case does not exist because in the source code of JAX, the type of `params` in `JaxEqn` is marked as `Any`. May I treat it as constant in this PR and deal with it if meeting non-constant case in the future?

Reply:
leave it for now as is, but if there is really always a constant you could use the `get_attribute` API that is also used in TF and other FEs.